starter-code/src/App.js

```diff
@@ -1,19 +1,92 @@
 import React, { Component } from 'react';
-import logo from './logo.svg';
 import './App.css';
+import 'bulma/css/bulma.css';
+import foods from './foods.json';
+import FoodBox from './FoodBox';
+import FormFood from './FormFood';
+import SearchBar from './SearchBar';
+import TodaysFoods from './TodaysFoods';
+
 
 class App extends Component {
+  state = {
+    allFoods: foods,
+    // todaysFoods pour pouvoir avoir un tableau avec nos datas à afficher
+    todaysFoods: [],
+    showForm: false,
+  }
+
+  addFood = newFood => {
+    const copyFoods = [...this.state.allFoods];
+    copyFoods.push(newFood)
+    this.setState({ allFoods: copyFoods });
+
+  }
+
+  displayForm = () => {
+    this.setState({ showForm: !this.state.showForm });
+  }
+
+  handleSearch = (value) => {
+    let filteredFood = foods.filter(food => food.name.toUpperCase().includes(value.toUpperCase()));
+    this.setState({ allFoods: filteredFood });
+  }
+
+  // 4) fonction handleAddFood déclarée ici avec ses paramètres (qu'elle recoit de onAdd dans composant Foodbox via addToday)
+  handleAddFood = (name, quantity, calories) => {
+    const todaysFoodsState = this.state.todaysFoods;
+    const food = { name, quantity, calories };
+
+    todaysFoodsState.push(food);
+
+    this.setState({
+      todaysFoods: todaysFoodsState
+    });
+  }
+
+
   render() {
     return (
       <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
-        </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
-      </div>
+        <h1>IronNutrition</h1>
+
+        <button
+          className="add-food"
+          onClick={this.displayForm}
+        >add food</button>
+
+        {/* affiche le formulaire pour add a food */}
+        <FormFood isVisible={this.state.showForm} clbk={this.addFood} handleDisplayForm={this.displayForm} />
+
+        {/* affiche la searchBar et filtre */}
+        <SearchBar clbk={this.handleSearch} />
+
+
+        <div className="foods-and-todays">
+          <div className="all-foods-list">
+            {/* affiche tous les foods */}
+            {
+              this.state.allFoods.map((food, index) =>
```

Comment:
you should better filter you list before the map: this.state.foods.filter(value of the search).map()
Thus you don't loose your new food.

And if the search is empty, it will display all the food.